File: src/main/java/hilfling/backend/hilfling/model/PhotoGangBanger.java
package hilfling.backend.hilfling.model;

import lombok.Data;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.validator.constraints.URL;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;
import javax.validation.constraints.Email;

@Data
@Entity(name = "photo_gang_bangers")
@Table(name = "photo_gang_bangers")
public class PhotoGangBanger extends User {
    public PhotoGangBanger() {
    }

    public PhotoGangBanger(
            String firstName,
            String lastName,
            String username,
            @Email String email,
            @URL String profilePicture,
            String phoneNumber,
            String sex,
            String relationshipStatus,
            String semesterStart,
            Boolean active,
            Boolean pang,
            String address,
            String zipCode,
            String city
    ) {
        super(firstName, lastName, username, email, profilePicture, phoneNumber, sex);
        this.relationshipStatus = relationshipStatus;
        this.semesterStart = semesterStart;
        this.active = active;
        this.pang = pang;
        this.address = address;
        this.zipCode = zipCode;
        this.city = city;
    }

    @Column(name = "relationship_status", nullable = false)
    private String relationshipStatus;

    @Column(name = "semester_start", nullable = false)
    private String semesterStart;

    @Column(name = "active", nullable = false)
    @ColumnDefault("true")
    private Boolean active;

    @Column(name = "pang", nullable = false)
    @ColumnDefault("false")
    private Boolean pang;

    @Column(name = "address", nullable = false)
    private String address;

    @Column(name = "zip_code", nullable = false)
    private String zipCode;

    @Column(name = "city", nullable = false)
    private String city;
}

File: src/main/kotlin/hilfling/backend/hilfling/model/PhotoGangBanger.kt
package hilfling.backend.hilfing.model

import me.liuwj.ktorm.database.Database
import me.liuwj.ktorm.entity.Entity
import me.liuwj.ktorm.entity.sequenceOf
import me.liuwj.ktorm.schema.*

interface PhotoGangBanger : BaseModel<PhotoGangBanger> {
    companion object : Entity.Factory<PhotoGangBanger>()

    var relationshipStatus: String
    var semesterStart: String
    var isActive: Boolean
    var isPang: Boolean
    var address: String
    var zipCode: String
    var city: String

    // From User model
    var user: User
}

object PhotoGangBangers : BaseTable<PhotoGangBanger>("photo_gang_bangers") {
    val relationshipStatus = varchar("relationship_status").bindTo { it.relationshipStatus }
    val semesterStart = varchar("semester_start").bindTo { it.semesterStart }
    val isActive = boolean("is_active").bindTo { it.isActive }
    val isPang = boolean("is_pang").bindTo { it.isPang }
    val address = varchar("address").bindTo { it.address }
    val zipCode = varchar("zip_code").bindTo { it.zipCode }
    val city = varchar("city").bindTo { it.city }

    // From User model
    val userId = int("user_id").references(Users){it.user}
}

val Database.photo_gang_bangers get() = this.sequenceOf(PhotoGangBangers)